packages/react-devtools-shared/src/__tests__/HooksTree-test.js

```diff
@@ -0,0 +1,92 @@
+describe('HooksTree', () => {
+  let React;
+  let ReactDOM;
+  let bridge: FrontendBridge;
+  let store: Store;
+  let utils;
+  let HooksTree;
+
+  let BridgeContext;
+  let InspectedElementContextController;
+  let StoreContext;
+  let TreeContextController;
+
+  beforeEach(() => {
+    utils = require('./utils');
+    utils.beforeEachProfiling();
+
+    bridge = global.bridge;
+    store = global.store;
+    store.collapseNodesByDefault = false;
+
+    React = require('react');
+    ReactDOM = require('react-dom');
+
+    BridgeContext = require('react-devtools-shared/src/devtools/views/context')
+      .BridgeContext;
+    InspectedElementContextController = require('react-devtools-shared/src/devtools/views/Components/InspectedElementContext')
+      .InspectedElementContextController;
+    StoreContext = require('react-devtools-shared/src/devtools/views/context')
+      .StoreContext;
+    TreeContextController = require('react-devtools-shared/src/devtools/views/Components/TreeContext')
+      .TreeContextController;
+
+    HooksTree = require('react-devtools-shared/src/devtools/views/Components/HooksTree')
+      .HooksTreeView;
+  });
+
+  const Contexts = ({
+    children,
+    defaultSelectedElementID = null,
+    defaultSelectedElementIndex = null,
+  }) => (
+    <BridgeContext.Provider value={bridge}>
+      <StoreContext.Provider value={store}>
+        <TreeContextController
+          defaultSelectedElementID={defaultSelectedElementID}
+          defaultSelectedElementIndex={defaultSelectedElementIndex}>
+          <InspectedElementContextController>
+            {children}
+          </InspectedElementContextController>
+        </TreeContextController>
+      </StoreContext.Provider>
+    </BridgeContext.Provider>
+  );
+
+  it('shows the complex value of custom hooks with sub-hooks', () => {
+    // props from `InspectedElementContext display complex values of useDebugValue: DisplayedComplexValue 1` snapshot
```

Comment:
To make it more explicit I could merge the added test from `inspectedElementContext` and instead of serializing the tree directly pass `inspectedElement.hooks` to the tested `HooksTree`. A core member should decide if this coupling should be explicit or not.